src/services/completions.ts

```diff
@@ -2216,14 +2216,9 @@ namespace ts.Completions {
                 if (canGetType) {
                     const typeForObject = typeChecker.getTypeAtLocation(objectLikeContainer);
                     if (!typeForObject) return GlobalsSearch.Fail;
-                    // In a binding pattern, get only known properties (unless in the same scope).
-                    // Everywhere else we will get all possible properties.
-                    const containerClass = getContainingClass(objectLikeContainer);
-                    typeMembers = typeChecker.getPropertiesOfType(typeForObject).filter(symbol =>
-                        // either public
-                        !(getDeclarationModifierFlagsFromSymbol(symbol) & ModifierFlags.NonPublicAccessibilityModifier)
-                        // or we're in it
-                        || containerClass && contains(typeForObject.symbol.declarations, containerClass));
+                    typeMembers = typeChecker.getPropertiesOfType(typeForObject).filter(propertySymbol => {

```

Comment:
this code is pretty checker-heavy. Maybe it would make sense to have

```ts
function getAccessiblePropertiesOfTypeAtLocation(location: node, isSuper: boolean, writing: boolean, type: Type)
```

Also I'm not sure when `isSuper` or `writing` might be true, so it might be possible to drop them or calculate them in the above function.

Reply:
could this make the function less general/useful? it seems like a good idea to me, but I'm not familiar enough with the codebase to know if there are other places this function could be used, either as is or as `getAccessiblePropertiesOfTypeAtLocation` as you suggested

Reply:
I'm undecided myself. I was mainly thinking that it would make both the signature and the code smaller. (But it's not *always* a good idea to use Kolmogorov complexity to guide development. =)